File: src/server/src/main/java/com/project/application/models/user/UserViewResource.java
package com.project.application.models.user;

import com.project.domain.user.User;
import io.micronaut.serde.annotation.Serdeable;

import java.util.UUID;

@Serdeable
public record UserViewResource(UUID id, String fullName, String email) {
    public static UserViewResource from(User user) {
        return new UserViewResource(user.getId(), user.getFullName(), user.getEmail());
    }
}

File: src/server/src/main/java/com/project/application/models/user/UserViewResource.kt
package com.project.application.models.user

import com.project.domain.user.User
import io.micronaut.serde.annotation.Serdeable
import java.util.UUID

@Serdeable
data class UserViewResource(val id: UUID, val fullName: String, val email: String?) {
    companion object {
        fun from(user: User): UserViewResource {
            return UserViewResource(user.id, user.getFullName(), user.email)
        }
    }
}
